src/nomad_simulations/model_method.py

```diff
@@ -392,6 +402,209 @@ def normalize(self, archive, logger) -> None:
+class PerturbationMethod(ModelMethodElectronic):
+    type = Quantity(
+        type=MEnum('MP', 'RS', 'BW'),
+        description="""
+        Perturbation approach. The abbreviations stand for:
+        | Abbreviation | Description |
+        | ------------ | ----------- |
+        | `'MP'`       | Moller-Plesset |
+        | `'RS'`       | Rayleigh-Schrödigner |
+        | `'BW'`       | Brillouin-Wigner |
+        """,
+        a_eln=ELNAnnotation(component='EnumEditQuantity'),
+    )  # TODO: check if the special symbols are supported
+
+    order = Quantity(
+        type=np.int32,
+        description="""
+        Order up to which the perturbation is expanded.
+        """,
+        a_eln=ELNAnnotation(component='NumberEditQuantity'),
+    )
+
+    is_size_extensive = Quantity(
+        type=bool,
+        description="""
+        If the perturbation method is size extensive.
+        """,
+        a_eln=ELNAnnotation(component='BoolEditQuantity'),
+    )  # this should go into a knowledge base
+
+
+class CoupledCluster(ModelMethodElectronic):
+    """
+    A base section used to define the parameters of a Coupled Cluster calculation.
+    A standard schema is defined, though the most common cases can be summarized in the `type` quantity.
+    Conversion between both is provided.
+    """
+
+    order_map = {k: v for k, v in enumerate(('S', 'D', 'T', 'Q'))}
+    solver_map = {'QV': 'quasi-variational', 'B': 'Brueckner'}
+    map_solver = {v: k for k, v in solver_map.items()}
+
+    type = Quantity(
+        type=MEnum(
+            *[
+                f'{prefix}CC{base_order}{triples}{quadruples}{corr}'
+                for prefix in ('', 'QV', 'B')
+                for base_order in ('', 'D', 'SD')
+                for triples in ('', '(T)', 'T')
+                for quadruples in ('', '(Q)', 'Q')
+                for corr in ('', '-F12', '-R12')
+            ]
+        ),
+        description="""
+        Coupled Cluster method.
+        """,
+        a_eln=ELNAnnotation(component='EnumEditQuantity'),
+    )  # TODO: is too generous, remove non-sensical combinations
+
+    excitation_order = Quantity(
+        type=np.int32,
+        shape=['*'],
+        description="""
+        Orders at which the excitation are used.
+        1 = single, 2 = double, 3 = triple, 4 = quadruple, etc.
+
+        Note that coupled cluster typically start from doubles.
+        Singles excitations in a Koopman-compliant scheme only make sense as a response to a perturbation.
+        """,
+        a_eln=ELNAnnotation(component='NumberEditQuantity'),
+    )
+
+    perturbative_order = Quantity(
+        type=np.int32,
+        shape=['*'],
+        description="""
+        Excitation order at which the perturbative correction is used.
+        1 = single, 2 = double, 3 = triple, 4 = quadruple, etc.
+        """,
+        a_eln=ELNAnnotation(component='NumberEditQuantity'),
+    )
+
+    perturbative_method = SubSection(sub_section=PerturbationMethod.m_def)
```

Comment:
You are also mixing here "Perturbative" and "Perturbation". Here I'll use also `Perturbation`.